File: baekjoon/도미노_무너트리기.js
const fs = require("fs");
const [num, ...cases] = fs.readFileSync("/dev/stdin").toString().split("\n");

const N = +num;

cases.sort((a, b) => {
  [aX, aL] = a.split(" ").map(Number);
  [bX, bL] = b.split(" ").map(Number);
  return aX - bX;
});

let count = 0;
let lastX = 0;
for (let i = 0; i < N; i++) {
  const [x, l] = cases[i].split(" ").map(Number);
  if (lastX < x) {
    count++;
  }
  lastX = x + l;
}

console.log(count);

File: baekjoon/도미노_무너트리기.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"sort"
	"strconv"
	"strings"
)

type Domino struct {
	x, length int
}

func main() {
	scanner := bufio.NewScanner(os.Stdin)

	scanner.Scan()
	n, _ := strconv.Atoi(scanner.Text())

	domino := make([]Domino, n)
	for i := 0; i < n; i++ {
		scanner.Scan()
		input := strings.Split(scanner.Text(), " ")
		x, _ := strconv.Atoi(input[0])
		length, _ := strconv.Atoi(input[1])
		domino[i] = Domino{x, length}
	}

	sort.Slice(domino, func(i, j int) bool {
		return domino[i].x < domino[j].x
	})

	count := 1

	for i := 0; i < n-1; i++ {
		if domino[i].x+domino[i].length < domino[i+1].x {
			count++
		}
	}

	fmt.Println(count)
}
